File: 1404_Number_of_Steps_to_Reduce_a_Number_in_Binary_Representation_to_One/solution.c
int numSteps(char* s) {
    int r = 0, c = 0;
    for (int i = strlen(s) - 1; i > 0; i--) { int b = s[i] & 1; r += 1 + (b ^ c); c |= b; }
    return r + c;
}

File: 1404_Number_of_Steps_to_Reduce_a_Number_in_Binary_Representation_to_One/solution.py
class Solution:
    def numSteps(self, s: str) -> int:
        st, c = 0, 0
        for i in range(len(s) - 1, 0, -1):
            b = ord(s[i]) & 1
            st += 1 + (b ^ c)
            c |= b
        return st + c